lib/Cron/Workers/InactiveSubscribers.php

```diff
@@ -28,26 +28,31 @@ function __construct(
 
 
   function processTaskStrategy(ScheduledTask $task) {
-    $tracking_enabled = (bool)$this->settings->get('tracking.enabled');
-    if (!$tracking_enabled) {
+    try {
+      $tracking_enabled = (bool)$this->settings->get('tracking.enabled');
+      if (!$tracking_enabled) {
+        self::schedule();
+        return true;
+      }
+      $days_to_inactive = (int)$this->settings->get('deactivate_subscriber_after_inactive_days');
+      // Activate all inactive subscribers in case the feature is turned off
+      if ($days_to_inactive === 0) {
+        $this->inactive_subscribers_controller->reactivateInactiveSubscribers();
+        self::schedule();
+        return true;
+      }
+      // Handle activation/deactivation within interval
+      while ($this->inactive_subscribers_controller->markInactiveSubscribers($days_to_inactive, self::BATCH_SIZE) === self::BATCH_SIZE) {
+        CronHelper::enforceExecutionLimit($this->timer);
+      };
+      while ($this->inactive_subscribers_controller->markActiveSubscribers($days_to_inactive, self::BATCH_SIZE) === self::BATCH_SIZE) {
+        CronHelper::enforceExecutionLimit($this->timer);
+      };
       self::schedule();
-      return true;
+    } catch (\Exception $e) {
```

Comment:
I think it would be more readable to move the wrapping into try/catch block one level up into `processTask`. I see exactly same try/catch with same error handling in `WooCommerceSync` worker.
If we move it in process task and wrap `processTaskStrategy` we could easily see what happens in case of exception directly from the processTask method which is the one which is handling the "administration" stuff of task processing.